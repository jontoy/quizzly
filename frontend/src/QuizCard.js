import React from "react";
import "./QuizCard.css";

const QuizCard = ({ id, name, difficulty }) => {
  return (
    <div className="QuizCard card p-2 mb-1">
      <h5 className="card-title">{name}</h5>
      <p className="card-text">Difficulty: {difficulty}</p>
    </div>
  );
};

export default QuizCard;
