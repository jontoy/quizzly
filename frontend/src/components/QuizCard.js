import React from "react";
import "./QuizCard.css";

const QuizCard = ({ name, difficulty }) => {
  return (
    <div className="QuizCard card p-2 mb-1 bg-light">
      <h5 className="card-title">{name}</h5>
      <p className="card-text">Difficulty: {difficulty}</p>
    </div>
  );
};

export default React.memo(QuizCard);
