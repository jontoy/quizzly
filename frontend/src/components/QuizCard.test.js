import React from "react";
import { render } from "@testing-library/react";
import QuizCard from "./QuizCard";

it("renders without crashing", () => {
  render(<QuizCard />);
});

it("matches snapshot", () => {
  const { asFragment } = render(<QuizCard />);
  expect(asFragment()).toMatchSnapshot();
});
