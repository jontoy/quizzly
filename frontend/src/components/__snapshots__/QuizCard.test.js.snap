// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`matches snapshot 1`] = `
<DocumentFragment>
  <div
    class="QuizCard card p-2 mb-1 bg-light"
  >
    <h5
      class="card-title"
    />
    <p
      class="card-text"
    >
      Difficulty: 
    </p>
  </div>
</DocumentFragment>
`;
